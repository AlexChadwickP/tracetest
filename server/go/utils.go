package openapi

import (
	"net/http"

	"github.com/gogo/protobuf/jsonpb"
	"github.com/gogo/protobuf/proto"
	v11 "go.opentelemetry.io/proto/otlp/common/v1"
	res "go.opentelemetry.io/proto/otlp/resource/v1"
	v1 "go.opentelemetry.io/proto/otlp/trace/v1"
)

func EncodeJSONPBResponse(i interface{}, status *int, w http.ResponseWriter) error {
	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
	if status != nil {
		w.WriteHeader(*status)
	} else {
		w.WriteHeader(http.StatusOK)
	}

	trace := i.(proto.Message)
	m := jsonpb.Marshaler{}
	return m.Marshal(w, trace)
}

func FixParent(tr *v1.TracesData, traceID, parentSpanID string) *v1.TracesData {
	spans := make(map[string]*v1.Span)
	for _, rs := range tr.ResourceSpans {
		for _, ils := range rs.InstrumentationLibrarySpans {
			for _, sp := range ils.Spans {
				spans[string(sp.SpanId)] = sp
			}
		}
	}

	// Fix parent id
	for _, sp := range spans {
		if sp.ParentSpanId == nil {
			continue
		}
		_, ok := spans[string(sp.ParentSpanId)]
		if !ok {
			sp.ParentSpanId = []byte(parentSpanID)
		}
	}
	rs := &v1.ResourceSpans{
		Resource: &res.Resource{
			Attributes:             []*v11.KeyValue{},
			DroppedAttributesCount: 0,
		},
		InstrumentationLibrarySpans: []*v1.InstrumentationLibrarySpans{
			{
				InstrumentationLibrary: &v11.InstrumentationLibrary{
					Name:    "tracetest",
					Version: "v1",
				},
				Spans: []*v1.Span{
					{TraceId: []byte(traceID),
						SpanId:       []byte(parentSpanID),
						ParentSpanId: nil,
						Name:         "tracetest",
						Kind:         v1.Span_SPAN_KIND_CLIENT,
					},
				},
			},
		},
		SchemaUrl: "",
	}
	tr.ResourceSpans = append(tr.ResourceSpans, rs)

	return tr
}
