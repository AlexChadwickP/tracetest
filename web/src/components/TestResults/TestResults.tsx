import SkeletonTable from 'components/SkeletonTable';
import {FC} from 'react';
import {TTestRun} from '../../types/TestRun.types';
import AssertionCardList from '../AssertionCardList';

type TTestResultsProps = {
  onSelectSpan(spanId: string): void;
  testId: string;
  run: TTestRun;
};

const TestResults: FC<TTestResultsProps> = ({run: {trace, result}, testId, onSelectSpan}) => {
  return (
    <SkeletonTable loading={!trace}>
      <AssertionCardList assertionResults={result} onSelectSpan={onSelectSpan} testId={testId} />
    </SkeletonTable>
  );
};

export default TestResults;
