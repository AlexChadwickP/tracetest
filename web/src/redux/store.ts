import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import {testAPI} from '../services/TestService';

export const store = configureStore({
  reducer: {
    [testAPI.reducerPath]: testAPI.reducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
