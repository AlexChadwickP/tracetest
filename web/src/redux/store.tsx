import {Action, configureStore, ThunkAction} from '@reduxjs/toolkit';
import TestAPI from 'redux/apis/TraceTest.api';

export const store = configureStore({
  reducer: {
    [TestAPI.reducerPath]: TestAPI.reducer,
  },
  middleware: getDefaultMiddleware => getDefaultMiddleware().concat(TestAPI.middleware),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
