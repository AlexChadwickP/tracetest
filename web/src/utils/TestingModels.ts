export const TestingModels = {
  assertionResult: {
    assertion: {assertionId: '', selectors: [], spanAssertions: []},
    spanListAssertionResult: [],
  },
};
