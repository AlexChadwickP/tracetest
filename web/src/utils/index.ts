const flattenAttributesSelector = () =>
  `resourceSpans[].[instrumentationLibrarySpans[].spans[].attributes[].{key:key,value:value.*|[0]},resource.attributes[].{key:key,value: value.*|[0]}]|[][]`;

export const filterBySpanId = (spanId: string = '') =>
  `resourceSpans[?instrumentationLibrarySpans[?spans[?starts_with(spanId,'${spanId}')]]] | [].[instrumentationLibrarySpans[].spans[].attributes[].{key:key,value:value.*|[0],type:'span'},resource.attributes[].{key:key,value: value.*|[0],type:'resource'}]|[][]`;
