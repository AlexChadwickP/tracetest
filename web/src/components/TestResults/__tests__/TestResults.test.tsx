import {render} from '@testing-library/react';
import TestResults from '../TestResults';
import {TestingModels} from '../../../utils/TestingModels';
import {ReduxWrapperProvider} from '../../../redux/ReduxWrapperProvider';

test('SpanAttributesTable', () => {
  const setMax = jest.fn();
  const onSpanSelected = jest.fn();
  const onPointerDown = jest.fn();
  const setHeight = jest.fn();
  const result = render(
    <ReduxWrapperProvider>
      <TestResults
        result={TestingModels.testRunResult}
        visiblePortion={0}
        max={0}
        setMax={setMax}
        onSpanSelected={onSpanSelected}
        height={100}
        onPointerDown={onPointerDown}
        setHeight={setHeight}
      />
    </ReduxWrapperProvider>
  );
  expect(result.container).toMatchSnapshot();
});
