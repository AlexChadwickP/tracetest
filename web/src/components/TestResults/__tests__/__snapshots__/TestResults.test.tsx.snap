// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpanAttributesTable 1`] = `
<div>
  <div
    class="sc-eCImPb igWlOZ"
    data-cy="test-results"
  >
    <div
      class="sc-bdvvtL ftmYax"
      data-tour="trace_timeline"
      style="height: 0px;"
    >
      <span
        class="ant-typography sc-gsDKAQ"
      >
        <strong>
          Trace Information
        </strong>
      </span>
      <span
        class="ant-typography sc-dkPtRN hoDAHZ"
      >
        Trace Start 
        Tuesday, 17th May 2022 at 09:38:04
      </span>
      <span
        class="ant-typography sc-gsDKAQ"
      >
        <strong>
          1
           total spans • 
          0
           selectors • 
          0
           checks • 
          <span
            class="ant-badge ant-badge-not-a-wrapper"
          >
            <sup
              class="ant-scroll-number ant-badge-count"
              data-show="true"
              style="background-color: rgb(73, 170, 25);"
              title="P"
            >
              P
            </sup>
          </span>
           
          <span
            class="sc-hKwDye lhDCTR"
          >
            0
          </span>
          <span
            class="ant-badge ant-badge-not-a-wrapper"
          >
            <sup
              class="ant-scroll-number ant-badge-count"
              data-show="true"
              title="F"
            >
              F
            </sup>
          </span>
           
          <span
            class="sc-hKwDye lhDCTR"
          >
            0
          </span>
        </strong>
      </span>
    </div>
    <div
      class="sc-iCfMLu cmTlVF"
    >
      <div>
        <div
          class="sc-jRQBWg kKtPgd"
        >
          <img
            class="sc-gKclnd"
            src="SpanAssertionsEmptyState.svg"
          />
          <span
            class="ant-typography ant-typography-disabled"
          >
            No Data
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
